master/buildbot/test/util/integration.py

```diff
@@ -50,7 +49,41 @@ def loadConfig(self):
         return MasterConfig.loadFromDict(self.config_dict, '<dict>')
 
 
-class RunMasterBase(dirs.DirsMixin, unittest.TestCase):
+@defer.inlineCallbacks
+def getMaster(case, reactor, config_dict):
+    """
+    Create a started ``BuildMaster`` with the given configuration.
+    """
+    basedir = FilePath(case.mktemp())
+    basedir.createDirectory()
+    master = BuildMaster(basedir.path, reactor=reactor, config_loader=DictLoader(config_dict))
+
+    # TODO: Allow BuildMaster to transparently upgrade the database, at least for tests.
+    yield master.db.setup(check_version=False)
+    yield master.db.model.upgrade()
+    master.db.setup = lambda: None
+
+    yield master.startService()
+
+    defer.returnValue(master)
+
+
+def getBuilderIdByName(master, builder_name):
+    """
+    Get the ``builder_id`` of the given builder.
+    """
+    d = master.data.get(('builders',))
```

Comment:
> This code is essentially how the schedulers get that information currently.
Really? arg, that is not scalable.

I agree that there should be a data api binding findBuilderId.. 

actually, there is one.

master.data.updates.findBuilderId(name)